server/async_tcp.go

```diff
@@ -17,247 +16,211 @@ import (
 	"github.com/dicedb/dice/server/utils"
 )
 
-var cronFrequency time.Duration = 1 * time.Second
-var lastCronExecTime time.Time = utils.GetCurrentTime()
-
-const EngineStatusWAITING int32 = 1 << 1
-const EngineStatusBUSY int32 = 1 << 2
-const EngineStatusSHUTTINGDOWN int32 = 1 << 3
-const EngineStatusTRANSACTION int32 = 1 << 4
-
-var eStatus int32 = EngineStatusWAITING
-
-var connectedClients map[int]*core.Client
+type AsyncServer struct {
+	serverFD         int
+	ctx              context.Context
+	cancel           context.CancelFunc
+	wg               *sync.WaitGroup
+	maxClients       int
+	multiplexer      iomultiplexer.IOMultiplexer
+	connectedClients map[int]*core.Client
+	store            *core.Store
+	lastCronExecTime time.Time
+	cronFrequency    time.Duration
+}
 
-var asyncStore = core.NewStore()
+// NewAsyncServer initializes a new AsyncServer
+func NewAsyncServer(wg *sync.WaitGroup) *AsyncServer {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
There is no need to create a context in the internal function, instead have it in the main and propagate it to each respective method associated with this function